File: backend/Testing/Browser/Page/UserAccountSettingsPage.cs
using Microsoft.Playwright;

namespace Testing.Browser.Page;

public class UserAccountSettingsPage : AuthenticatedBasePage<UserAccountSettingsPage>
{
    public UserAccountSettingsPage(IPage page)
    : base(page, "/user", page.GetByRole(AriaRole.Heading, new() { Name = "Account Settings" }))
    {
    }

    internal Task FillDisplayName(string name)
    {
        return Page.GetByLabel("Display name").FillAsync(name);
    }

    internal Task FillEmail(string email)
    {
        return Page.GetByLabel("Email").FillAsync(email);
    }

    internal async Task ClickSave()
    {
        await Page.GetByRole(AriaRole.Button, new() { Name = "Update account" }).ClickAsync();
        await Page.WaitForLoadStateAsync(LoadState.DOMContentLoaded);
    }

    internal async Task<ResetPasswordPage> ClickResetPassword()
    {
        await Page.GetByRole(AriaRole.Link, new() { Name = "Reset your password" }).ClickAsync();
        return await new ResetPasswordPage(Page).WaitFor();
    }
}

File: frontend/tests/pages/userAccountSettingsPage.ts
import type { Page } from '@playwright/test';
import { AuthenticatedBasePage } from './authenticatedBasePage';
import { ResetPasswordPage } from './resetPasswordPage';

export class UserAccountSettingsPage extends AuthenticatedBasePage {
  constructor(page: Page) {
    super(page, page.getByRole('heading', {name: 'Account Settings'}), `/user`);
  }

  fillDisplayName(name: string): Promise<void> {
    return this.page.locator('form').getByLabel('Display name').fill(name);
  }

  fillEmail(email: string): Promise<void> {
    return this.page.locator('form').getByLabel('Email').fill(email);
  }

  async clickSave(): Promise<void> {
    await this.page.getByRole('button', {name: 'Update account'}).click();
    await this.page.waitForLoadState('domcontentloaded');
  }

  async clickResetPassword(): Promise<ResetPasswordPage> {
    await this.page.getByRole('link', {name: 'Reset your password'}).click();
    return await new ResetPasswordPage(this.page).waitFor();
  }
}
